### ISSUE TITLE
Add support for block-level linking

### ISSUE THREAD

**Issue by robgrace** [foambubble/foam#294]
(This is more of a core feature, rather than an enhancement, inasmuch as foam is trying to replicate Roam behavior):

Roam, and now Obsidian, provide support for block-level linking between markdown notes.  It would be great if Roam could follow suit and provide this capapbilty.

Unfortunately, there are now three styles of intra-note linking to choose from:
- **Roam**:   ((block-id))      to text block (bullet)
- **Obsidian**:  [[NoteTitle#^block-id]]     -- link to a text block annotated with `^block-id`
- **Extended Markdown**:    [Description](#heading-id)     -- link to a heading annotated with `{#custom-id}`

I vote for the Obsidian-style linking, since this is the format that I'd imagine most people will be co-editing with vscode.

**Comment by ingalless** [comment]
Hey @robgrace!
Great suggestion, and one that had work started on it a while back:

[Foam Sneak Peek: Block Referencing and Block Embedding](https://youtu.be/ZqLEGGmOw-8)

It's one of the features we want in Foam

**Comment by 5hay** [comment]
I really like the `^hash` syntax at the end of a block. Is there any progress made or timeline specified when we can expect a first version that includes this feature? Have a nice Sunday everyone.

**Comment by kisnows** [comment]
Really need this feature.

**Comment by khramoff** [comment]
Hey, @ingalless. Definitely looks great! Is there any news, progress? 

Long time, no see. 
If there's any way, I can participate in this, it would be nice.

**Comment by ingalless** [comment]
@khramoff @5hay I've not heard of any progress recently. Foam has been undergoing a lot of underyling changes as of late. If I hear any whispers I'll try remember to update this 👍🏻

**Comment by Kcalb35** [comment]
I'm looking forward to this feature, because I really need to re-arrange my blocks of notes in another way. Hope to hear of any progress.

**Comment by ChenguangZhang** [comment]
It has been more than 1 year since the sneak peek video was uploaded and there is no block referencing (?). Any workaround?

**Comment by huyz** [comment]
> I vote for the Obsidian-style linking, since this is the format that I'd imagine most people will be co-editing with vscode.

+1.  Dendron also uses the `#^hash` link format.

**Comment by briansunter** [comment]
I would still really like to see this

**Comment by riccardoferretti** [comment]
Yes I agree, and the support for sections (#856) implicitly added support for blocks in the model (as a block can modeled as a "type" of section), now it would be a matter of changing the markdown parser to recognize the anchors.
Then of course we'd need to add any convenience that we'd want to make blocks easy to use (to be discussed).

**Comment by briansunter** [comment]
I also still want obsidian style block embed ![[foo]] from the video a year ago.

**Comment by yummta** [comment]
Any news about this? I came from Obsidian and this feature is a really useful.

**Comment by chrisUsick** [comment]
As an alternative perspective, I thought I’d share a way you can achieve
similar functionality in a foam-esque way.
Another option along side using sections is to create your blocks as
separate notes. Then, embed these “block notes using the embed syntax
`![[Note Name]]`.
This way you gain the ability to link to blocks, also your blocks will show
up in the graph and thus be more discoverable.
Just one more option. I know this wouldn’t work for everyone but I wanted
to share it.

**Comment by pderaaij** [comment]
I am wondering what's missing here. It looks like the requested feature is working. Although a minor issue needs to be fixed in the preview editor. 

Or do we want to be able to work more granular then sections?

**Comment by john-okeefe** [comment]
The one fix (or maybe checkbox option to select) is that Obsidian never released this feature as an in-line or seamless block. I would love to see that here.

**Comment by dongguaguaguagua** [comment]
Hey guys! I am wondering if there is any progress on this issue, or is it possible to add this feature? The [block reference in Obsidian](https://help.obsidian.md/Linking+notes+and+files/Internal+links#Link+to+a+block+in+a+note) is really helpful.

**Comment by tctlrd** [comment]
@ingalless who has the know how to make this happen?

**Comment by riccardoferretti** [comment]
happy to finally close this ticket ;) full block id support has been implemented as of 0.32.0 🚀
